build_tools/hack/env_check/check_tools.py

```diff
@@ -0,0 +1,875 @@
+from __future__ import annotations
+from typing import Literal, Optional, Union, Tuple
+from types import EllipsisType, NoneType
+import os, re, sys, shutil
+from pathlib import Path
+from abc import ABC
+from .utils import cstring, Emoji
+from .find_tools import *
+from .device import Device
```

Comment:
nit: some of these imports are not used. Editors like VSCode can highlight for you, and some tools may help remove unused import automatically:

<img width="493" height="257" alt="image" src="https://github.com/user-attachments/assets/d407e037-00d1-40c3-b58a-7e89d647bf1c" />


```suggestion
from __future__ import annotations
from typing import Literal
import os
from pathlib import Path
from abc import ABC
from .utils import cstring, Emoji
from .find_tools import *
from .device import Device
```